nilearn/input_data/masker_validation.py

```diff
@@ -0,0 +1,43 @@
+import warnings
+
+from .._utils.class_inspect import get_params
+from . import MultiNiftiMasker, NiftiMasker
+
+
+def check_embedded_nifti_masker(instance, multi=True):
+    """Base class for decomposition estimator. Takes care of mask logic and of transformation
+
+    Parameters
+    ==========
+        instance: object, instance of BaseEstimator
+            The object that gives us the values of the parameters
+
+        multi: boolean, default=True,
+            Indicates whether to return a MultiNiftiMasker or a NiftiMasker
+
+    Returns
+    ==========
+        masker: MultiNiftiMasker, NiftiMasker
+            New masker
+    """
+    masker_type = MultiNiftiMasker if multi else NiftiMasker
+    parameters = get_params(masker_type, instance)
+    mask = getattr(instance, 'mask', None)
+
+    if isinstance(mask, (NiftiMasker, MultiNiftiMasker)):
+        # Creating MultiNiftiMasker from mask
+        masker = masker_type(**get_params(masker_type, mask))
+        # Raising warning if providing user has provided a masker
+        for param_key in masker.get_params():
+            if param_key in parameters and getattr(masker, param_key) != parameters[param_key]:
+                warnings.warn("Provided mask overrides default/provided parameter %s" % param_key)
+        # Copying potential attribute
+        if hasattr(mask, 'mask_img_'):
+            # Allow free fit of returned mask
+            masker.mask_img = mask.mask_img_
+    else:
+        # Creating a MultiNiftiMasker with parameters extracted from instance
+        parameters.pop('mask_img', None)
```

Comment:
In case instance has both mask and mask_img parameters. I will remove it so that it raises ArgumentError in this case